winpcap: Missing headers in i686-w64-mingw32.static

I just tried to build https://dev.cloudtrax.com/git/ap51-flash.git but it failed because the winpcap package neither provides pcap.h nor Win32-Extensions.h and no *wpcap.(so|a). Seems to be a bug in the MXE winpcap package because it works with the WpdPack_4_1_2.zip

    make PLATFORM=WIN32 CROSS=i686-w64-mingw32.shared- V=s ap51-flash
    i686-w64-mingw32.shared-gcc -Wall -Werror -W -g3 -std=gnu99 -Os -fno-strict-aliasing -DWIN32 -D_GNU_SOURCE -DREVISION_VERSION=\"f94f9c9\"  -MD -c flash.c -o flash.o
    In file included from flash.h:22:0,
                     from flash.c:29:
    compat.h:64:18: fatal error: pcap.h: No such file or directory
    compilation terminated.
    Makefile:142: recipe for target 'flash.o' failed

The only files I could find in mxe were

* ./usr/i686-w64-mingw32.shared/include/wpcapi.h
* ./usr/i686-w64-mingw32.shared/include/wpcapi.idl

Not sure what I should to with them

Thanks for the fast reply. But also the shared version doesn't work. There seems to be a different problem but there is still a problem:

    $ make PLATFORM=WIN32 CROSS=i686-w64-mingw32.static- V=s ap51-flash
    i686-w64-mingw32.static-gcc -Wall -Werror -W -g3 -std=gnu99 -Os -fno-strict-aliasing -DWIN32 -D_GNU_SOURCE -DREVISION_VERSION=\"3f67302\"  -MD -c flash.c -o flash.o
    In file included from compat.h:64:0,
                     from flash.h:22,
                     from flash.c:29:
    /mxe/usr/i686-w64-mingw32.static/include/pcap.h:45:23: fatal error: pcap/pcap.h: No such file or directory
    compilation terminated.

There is now a more stuff in mxe but it seems that some of the required things are missing:

    $ find . -iname "*wpcap*"                                          
    ./pkg/WpcapSrc_4_1_3.zip
    ./usr/i686-w64-mingw32.static/lib/libwpcap.a
    ./usr/i686-w64-mingw32.static/include/WpcapNames.h
    ./usr/i686-w64-mingw32.static/include/wpcapi.h
    ./usr/i686-w64-mingw32.static/include/wpcapi.idl
    ./usr/i686-w64-mingw32.shared/include/wpcapi.h
    ./usr/i686-w64-mingw32.shared/include/wpcapi.idl
    $ find . -iname Win32-Extensions.h
    ./usr/i686-w64-mingw32.static/include/Win32-Extensions.h
    $ find . -iname pcap.h            
    ./usr/i686-w64-mingw32.static/include/pcap.h

Here is the interesting content from pcap.h:

    /*
     * For backwards compatibility.
     *
     * Note to OS vendors: do NOT get rid of this file!  Many applications
     * expect to be able to include <pcap.h>, and at least some of them
     * go through contortions in their configure scripts to try to detect
     * OSes that have "helpfully" moved pcap.h to <pcap/pcap.h> without
     * leaving behind a <pcap.h> file.
     */
    #include <pcap/pcap.h>

--- a/src/winpcap.mk
+++ b/src/winpcap.mk
@@ -22,8 +22,6 @@ define $(PKG)_BUILD
     cd '$(1)' && $(TARGET)-gcc -ICommon -IpacketNtx/Dll -O -c '$(1)/packetNtx/Dll/Packet32.c'
     $(TARGET)-ar rc '$(1)/libpacket.a' '$(1)/Packet32.o'
     $(TARGET)-ranlib '$(1)/libpacket.a'
-    $(INSTALL) -d '$(PREFIX)/$(TARGET)/include'
-    $(INSTALL) -m644 '$(1)/Common'/*.h '$(PREFIX)/$(TARGET)/include/'
     $(INSTALL) -d '$(PREFIX)/$(TARGET)/lib'
     $(INSTALL) -m644 '$(1)/libpacket.a' '$(PREFIX)/$(TARGET)/lib/'
 
@@ -34,7 +32,31 @@ define $(PKG)_BUILD
     AR='$(TARGET)-ar' \
     RANLIB='$(TARGET)-ranlib' \
     $(MAKE) -C '$(1)/wpcap/PRJ' -j 1 libwpcap.a
-    $(INSTALL) -m644 '$(1)/wpcap/libpcap/'*.h '$(1)/wpcap/Win32-Extensions/'*.h '$(PREFIX)/$(TARGET)/include/'
+
+    $(INSTALL) -d '$(PREFIX)/$(TARGET)/include'
+    $(INSTALL) -m644 \
+         $(1)/wpcap/libpcap/Win32/Include/IP6_misc.h \
+         $(1)/wpcap/libpcap/pcap-namedb.h \
+         $(1)/wpcap/libpcap/pcap-bpf.h \
+         $(1)/wpcap/libpcap/remote-ext.h \
+         $(1)/Common/Packet32.h \
+         $(1)/wpcap/Win32-Extensions/Win32-Extensions.h \
+         $(1)/wpcap/libpcap/Win32/Include/bittypes.h \
+         $(1)/wpcap/libpcap/pcap-stdinc.h \
+         $(1)/wpcap/libpcap/pcap.h \
+        '$(PREFIX)/$(TARGET)/include/'
+
+    $(INSTALL) -d '$(PREFIX)/$(TARGET)/include/pcap'
+    $(INSTALL) -m644 \
+        $(1)/wpcap/libpcap/pcap/namedb.h \
+        $(1)/wpcap/libpcap/pcap/sll.h \
+        $(1)/wpcap/libpcap/pcap/bpf.h \
+        $(1)/wpcap/libpcap/pcap/usb.h \
+        $(1)/wpcap/libpcap/pcap/pcap.h \
+        $(1)/wpcap/libpcap/pcap/bluetooth.h \
+        $(1)/wpcap/libpcap/pcap/vlan.h \
+        '$(PREFIX)/$(TARGET)/include/pcap/'
+
     $(INSTALL) -m644 '$(1)/wpcap/PRJ/libwpcap.a' '$(PREFIX)/$(TARGET)/lib/'
 endef
 

--- a/src/winpcap.mk
+++ b/src/winpcap.mk
@@ -20,7 +20,9 @@ endef
 define $(PKG)_BUILD
     # build
     cd '$(1)' && $(TARGET)-gcc -ICommon -IpacketNtx/Dll -O -c '$(1)/packetNtx/Dll/Packet32.c'
-    $(TARGET)-ar rc '$(1)/libpacket.a' '$(1)/Packet32.o'
+    cd '$(1)' && $(TARGET)-gcc -ICommon -IpacketNtx/Dll -O -c '$(1)/packetNtx/Dll/AdInfo.c'
+    cd '$(1)' && $(TARGET)-gcc -ICommon -IpacketNtx/Dll -O -c '$(1)/packetNtx/Dll/NpfImExt.c'
+    $(TARGET)-ar rc '$(1)/libpacket.a' '$(1)/Packet32.o' '$(1)/AdInfo.o' '$(1)/NpfImExt.o'
     $(TARGET)-ranlib '$(1)/libpacket.a'
     $(INSTALL) -d '$(PREFIX)/$(TARGET)/lib'
     $(INSTALL) -m644 '$(1)/libpacket.a' '$(PREFIX)/$(TARGET)/lib/'

